src/integrations/stripe.js

```diff
@@ -0,0 +1,101 @@
+import { loadStripe } from '@stripe/stripe-js';
+
+const stripePromise = new Promise((resolve) => {
+  document.addEventListener('stripe_key_loaded', (e) => {
+    const promise = loadStripe(e.detail.key);
+    promise.then((stripe) => {
+      resolve(stripe);
+    });
+  });
+});
+
+export function getStripePromise() {
+  return stripePromise;
+}
+export async function getStripe() {
+  return await getStripePromise();
+}
+
+export function installStripe(stripeConfig) {
+  // eslint-disable-next-line camelcase
+  if (stripeConfig?.metadata?.client_key) {
+    document.dispatchEvent(
+      new CustomEvent('stripe_key_loaded', {
+        detail: { key: stripeConfig.metadata.client_key }
+      })
+    );
+  }
+}
+
+export function getStripeCustomerFieldValues(stripeConfig, fieldValues) {
+  if (!stripeConfig?.metadata?.customer_field_mappings) return {};
+  const customer = getObjectMappingValues(
+    stripeConfig.metadata.customer_field_mappings,
+    fieldValues
+  );
+  // the shipping object in customer must have a name property (stripe requires this),
+  //  but we don't want to make the mapping required.  So default it here.
+  customer.shipping &&
```

Comment:
i would make this more explicit.

if (!customer.shipping?.name) customer.shipping.name = 'Name not supplied'

Reply:
There is a subtle but important difference between the code and what is suggested.  customer.shipping is not required, but if present, name must be there.  So changed it to be:
if (customer.shipping && !customer.shipping.name) customer.shipping.name = 'Name not supplied'